examples/analytics-iframe-transport-remote-frame.html

```diff
@@ -4,24 +4,26 @@
     <meta charset="UTF-8">
     <title>Requests Frame</title>
     <script>
-      window.addEventListener('amp-iframeTransportClientCreated', () => {
-        /**
-         * To receive AMP analytics events in a third-party frame, you must
-         * pass a callback function to this method. The callback will be called
-         * when an event is received, and will be passed two parameters:
-         * @param {string} event A string of the format specified in the
-         * requests block of the amp-analytics JSON config
-         * @param {string} transportId An ID uniquely identifying which creative
-         * generated the event
-         */
-        window.iframeTransportClient.onAnalyticsEvent(
-            (event, transportId) => {
-              // Now, do something meaningful with the AMP Analytics event
-              console.log('The page at: ' + window.location.href +
-                  ' has received an event: ' + event +
-                  ' from the creative with transport ID: ' + transportId);
-            });
-      });
+      /**
+       * To receive analytics events in a third-party frame, you must
+       * implement this method, with this signature. Within that method, you
+       * must create a function which processes an event, and
+       * pass that function to the onAnalyticsEvent() method of the object
+       * which is passed as a parameter to onNewAmpAnalyticsInstance().
+       * @param ampAnalytics  Call onAnalyticsEvent() on this, passing your
+       * function which will receive the analytics events.
+       */
+      window.onNewAmpAnalyticsInstance = ampAnalytics => {
```

Comment:
First: is `iframe: transport` in prod? I am afraid this can be a breaking change. 
Second: Can you explain to me why we change the interface to this earlier version? I get that we group the `onAnalyticsEvent` on creativeId. But I don't see an obvious advantage of doing this.
Third: Grouping listener on creativeId limit 3p vendor's ability to perform analyse operation between multiple creatives. (For example: check if more than one creative is currently in viewport). Is this something we care?

Reply:
1. It is in prod, but nobody has started using it yet :/
2. (Mentioned this in a chat, but pasting here for others)
```
So, at one point you asked, "I think vendor is fixed when creating the iframeTransportClient, can they specify vendor here?"
I agree with that - I think that my proposed code to specify vendor and creativeId as params to the method:
sendMessageToCreative(vendor, creativeId, response)
...is pretty ugly. In the original implementation, it was just
someObject.sendMessageToCreative(vendor)
...and the object knows all the necessary context info, so the customer doesn't have to keep track of it. keithwrightbos and I favored that all along.
In the old PR, lannka wanted it greatly abbreviated (iframe-transport-client.js was originally about 300 lines). So, he (lannka) suggested removing the object-orientedness, for brevity. But Keith and I feel that your question, plus the upcoming addition of other API stuff (intersectionObserver) mean we do need that context after all.
3. Interesting thought - this hasn't been discussed as a use case yet. But in that case, I think they could call some common function, such as:
```
      window.onNewAmpAnalyticsInstance = ampAnalytics => {
        ampAnalytics.onAnalyticsEvent(event => {
          if (...event indicates creative in viewport...) {
            incrementNumberOfCreativesInViewport();
          }
      });
    });

    function incrementNumberOfCreativesInViewport() {
      // No longer in the context of a single creative...
    }
```

Reply:
2. Sounds reasonable. Vendor should not worry about maintaining context instance for each creative. Do you still remember why @lannka wanted it greatly abbreviated (iframe-transport-client.js) though? 
3. The workaround sounds good. It's just not that intuitive to 3p vendor. I think we should really document well that each `onAnalyticsEvent` is for one creative instance then. 
In the meantime, the naming `onNewAmpAnalyticsInstance` and `ampAnalytics`  are quite confusing to me. It's actually a new creativeInstance or a new creativeContextInstance, and the instance is not ampAnalytics, but creative/context instead right?

Reply:
2. It was about 300 lines long - longer than the entire PR should be. The current version is 166 lines long.
3. Your suggested names SGTM.